### PR TITLE
fix: pending airdrop value for token with fractional fee and net of transfers false

### LINKED ISSUE #14896: Adding pending airdrop in to the state when token has fractional fee
If airdrop token with fractional fees and net of transfers is false, when it should be added into the pending state, 
after preparing the fee, the pending airdrop value should be reduced by the value of the fee.

### PR DESCRIPTION
**Pull request by JivkoKelchev**
**Description**:

Reduces the value of the pending airdrop, if the token has a fractional fee, with net of transfers false.

**Related issue(s)**:

Fixes #14896

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropBase.java [hunk 1]
```diff
@@ -64,11 +65,12 @@ public class TokenAirdropBase {
     protected static final String FT_WITH_HBAR_FIXED_FEE = "fungibleTokenWithHbarCustomFee";
     protected static final String FT_WITH_HTS_FIXED_FEE = "fungibleTokenWithHtsCustomFee";
     protected static final String FT_WITH_FRACTIONAL_FEE = "fungibleTokenWithFractionalFee";
+    protected static final String FT_WITH_FRACTIONAL_FEE_NET_OF_TRANSFERS = "ftWithFractionalFeeNetOfTransfers";
     protected static final String NFT_WITH_HTS_FIXED_FEE = "NftWithHtsFixedFee";
     protected static final String NFT_WITH_ROYALTY_FEE = "NftWithRoyaltyFee";
     protected static final String DENOM_TOKEN = "denomToken";
     protected static final String HTS_COLLECTOR = "htsCollector";
-    protected static final String HTS_COLLECTOR2 = "htsCollector2";
+    // protected static final String HTS_COLLECTOR2 = "htsCollector2";
```
Comment (ibankov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropBase.java:73]: We should delete this if we are not using it.

**Review by ibankov — DISMISSED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java [hunk 2]
```diff
@@ -134,9 +134,9 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         var convertedOp = CryptoTransferTransactionBody.newBuilder()
                 .tokenTransfers(op.tokenTransfers())
                 .build();
-        assessAndChargeCustomFee(context, convertedOp);
 
-        for (final var xfers : op.tokenTransfers()) {
+        var transfersAfterCustomFees = assessAndChargeCustomFee(context, convertedOp);
+        for (final var xfers : transfersAfterCustomFees.getFirst().tokenTransfers()) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:139]: Why are we doing `getFirst` ?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropBase.java (hunk 1 above)
Comment (JivkoKelchev) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropBase.java]: Done

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java]: Perfect. Thanks @JivkoKelchev

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM! Thanks @JivkoKelchev

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**